dynamo-db-client.js

```diff
@@ -199,28 +200,7 @@ class DynamoDbClient {
             ScanIndexForward: options.scanIndexForward,
             Select: options.select
         };
-
-        let itemsArray = [];
-        let additionalResultsData = [];
-        let results = {};
-        do {
-            if(results.LastEvaluatedKey != null) {
-                queryObject.ExclusiveStartKey = results.LastEvaluatedKey;
-            }
-            results = await this.docClient.query(queryObject).promise();
-            if(results.Items != null) {
-                itemsArray = itemsArray.concat(results.Items);
-            }
-            additionalResultsData.push({
-                count: results.Count,
-                scannedCount: results.ScannedCount,
-                consumedCapacity: results.ConsumedCapacity
-            });
-        } while (results.LastEvaluatedKey != null);
-        return {
-            items: itemsArray,
-            additionalResultsData
-        };
+        return runIterableMethod(this.docClient, 'query', queryOptions, options.doNotIterate);
```

Comment:
Any reason why not pass the method itself as parameter?

`return runIterableMethod(this.docClient.query, queryOptions, options.doNotIterate);`